src/OrbitGl/CaptureWindowTest.cpp

```diff
@@ -79,4 +81,193 @@ TEST_F(UnitTestCaptureWindow, SlidersRespondToMouseLeave) {
+        break;
+    }
+  }
+
+ private:
+  CaptureClientAppInterfaceFake capture_client_app_fake_;
+  std::unique_ptr<orbit_client_data::CaptureData> capture_data_;
+
+  void CreateSchedulerTimers() {
+    using orbit_client_protos::TimerInfo;
+    TimerInfo timer;
+    timer.set_start(0);
+    timer.set_end(100);
+    timer.set_thread_id(TrackTestData::kThreadId);
+    timer.set_processor(0);
+    timer.set_depth(0);
+    timer.set_type(TimerInfo::kCoreActivity);
+    time_graph_->ProcessTimer(timer, nullptr);
+  }
+};
+
+TEST_F(NavigationTestCaptureWindow, ZoomTimeWorksAsExpected) {
+  ExpectInitialState();
+
+  const Vec2i kTimeGraphSize = viewport_.WorldToScreen(time_graph_->GetSize());
+
+  // Zoom time: Zoom in and out horizontally, assure we've zoomed in, and zoomed out
+  // again
+  // Test 1: Middle of the capture window
+  int x = kTimeGraphSize[0] / 2;
+  int y = kTimeGraphSize[1] - 5;
+  MouseWheelMoved(x, y, 1, false);
+  ExpectIsHorizontallyZoomedIn(PosWithinCapture::kInBetween);
+  MouseWheelMoved(x, y, -1, false);
+  ExpectInitialState();
+
+  // Keyboard
+  MouseMoved(x, y, false, false, false);
+  KeyPressed('W', false, false, false);
+  ExpectIsHorizontallyZoomedIn(PosWithinCapture::kInBetween);
+  KeyPressed('S', false, false, false);
+  ExpectInitialState();
+
+  // Test 2: Left of the capture window
+  x = 0;
+  y = kTimeGraphSize[1] - 5;
+  MouseWheelMoved(x, y, 1, false);
+  ExpectIsHorizontallyZoomedIn(PosWithinCapture::kLeft);
+  MouseWheelMoved(x, y, -1, false);
+  ExpectInitialState();
+
+  // Keyboard
+  MouseMoved(x, y, false, false, false);
+  KeyPressed('W', false, false, false);
+  ExpectIsHorizontallyZoomedIn(PosWithinCapture::kLeft);
+  KeyPressed('S', false, false, false);
+  ExpectInitialState();
+
+  // Test 3: Right of time graph
+  x = kTimeGraphSize[0];
+  y = kTimeGraphSize[1] - 5;
+  MouseWheelMoved(x, y, 1, false);
+  ExpectIsHorizontallyZoomedIn(PosWithinCapture::kRight);
+  MouseWheelMoved(x, y, -1, false);
+  ExpectInitialState();
+
+  // Keyboard
+  MouseMoved(x, y, false, false, false);
+  KeyPressed('W', false, false, false);
+  ExpectIsHorizontallyZoomedIn(PosWithinCapture::kRight);
+  KeyPressed('S', false, false, false);
+  ExpectInitialState();
+}
+
+TEST_F(NavigationTestCaptureWindow, VerticalZoomWorksAsExpected) {
+  ExpectInitialState();
+
+  const Vec2i kTimeGraphSize = viewport_.WorldToScreen(time_graph_->GetSize());
+  int x = kTimeGraphSize[0] / 2;
+  int y = kTimeGraphSize[1] - 5;
+
+  float old_height = time_graph_->GetTrackContainer()->GetVisibleTracksTotalHeight();
+  MouseWheelMoved(x, y, 1, true);
+  UpdateChildrenPosAndSize();
+  EXPECT_GT(time_graph_->GetTrackContainer()->GetVisibleTracksTotalHeight(), old_height);
+
+  MouseWheelMoved(x, y, -1, true);
+  UpdateChildrenPosAndSize();
+  EXPECT_EQ(time_graph_->GetTrackContainer()->GetVisibleTracksTotalHeight(), old_height);
+}
+
+TEST_F(NavigationTestCaptureWindow, PanTimeWorksAsExpected) {
```

Comment:
I would also test in the right border or in the middle of the capture.

Reply:
Panning is independent of the mouse position, what would I test here?

Reply:
I mean, I would also test:
- Panning to the left doesn't change the state if you are in the leftmost part 
- idem to the right part.
- Test that moving left and after right (in a middle/right part of the screen) doesn't change the state.

Reply:
Ah, got it. That makes sense. I'm not sure I get that done this week, I would suggest that I add a bug / TODO here and add it in a different PR, wdyt?

Reply:
Added TODO